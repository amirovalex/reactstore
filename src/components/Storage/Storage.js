import React from 'react';
import Item from '../Item/Item';
// import Scroll from '../Scroll/Scroll';
import StorageItem from '../StorageItem/StorageItem';
import 'tachyons';

class Storage extends React.Component {

	constructor(props){
		super();
		this.state = {
			storage:[]
		}
	}

	componentDidMount() {
	fetch('https://still-escarpment-99159.herokuapp.com/getstorage')
		.then(response => response.json())
		.then(items => this.setState({storage:items}))
}

	render() {
		const { adminRoute , onChangeAdminRoute , onItemIdChange ,
				itemId , quantity , onGetItemQuantity ,
				isEditing , onItemEdit , onCancelEdit ,
				image , onEditQuantityXs ,onEditQuantityS , 
				onEditQuantityM , onEditQuantityL , onEditQuantityXl , 
				onEditItemImage} = this.props
		const { storage } = this.state
		const indexItem = storage.findIndex(item => item.itemid === itemId)
		console.log(indexItem)
		return(<div>
				{adminRoute === 'storage' ?
					(<div id="rows" className="store">
						{storage.length === 0 ?
							<p className="tc">The Storage is empty</p>
						:
						(storage.map((item,i)=> {
							return(
								<div key={i} onClick={() => {onChangeAdminRoute('storageitem');onItemIdChange(storage[i].itemid)}}>
								<Item 
								key={i}
								id={storage[i].itemid}
								name={storage[i].itemname}
								price={storage[i].price+'$'}
								category={storage[i].category}
								src={storage[i].itemimage}
								 />
								</div>) 
								}
							)
						)
						}
					</div>
					) : 
					(adminRoute === 'storageitem' ?
						<StorageItem
							onChangeAdminRoute = {onChangeAdminRoute}
				            onEditQuantityXs = {onEditQuantityXs}
				            onEditQuantityS = {onEditQuantityS}
				            onEditQuantityM = {onEditQuantityM}
				            onEditQuantityL = {onEditQuantityL}
				            onEditQuantityXl = {onEditQuantityXl}
				            onEditItemImage = {onEditItemImage}
							image={image}
							onCancelEdit={onCancelEdit}
							isEditing={isEditing}
							onItemEdit={onItemEdit} 
							quantity={quantity}
							onGetItemQuantity={onGetItemQuantity}
							itemId={itemId}
							storage={storage}
							src={storage[indexItem].itemimage}
							name={storage[indexItem].itemname}
							price={storage[indexItem].price}/>
						:
						null)
				}
				</div>
			)
	}
}

export default Storage;