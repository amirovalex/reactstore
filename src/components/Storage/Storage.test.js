import React from 'react';
import { shallow } from 'enzyme';
import Storage from './Storage.js';


it('expect to render component',() => {
	const mockStorage = [{
		category: "topwear",
		itemid: 1,
		itemimage: "https://cdn.rickowens.eu/products/69433/large/DU19F6285FEP8_09_M_2.jpg?1568643377",
		itemname: "Black Hoodie",
		price: 120
	}]
	const mockAdminRoute = 'storage'
	const mainComponent = shallow(<Storage/>)
	expect(mainComponent.debug()).toMatchSnapshot();
})

