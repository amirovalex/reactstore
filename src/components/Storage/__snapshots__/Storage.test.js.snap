// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render component 1`] = `"<div />"`;
